pkg/ipcache/ipcache.go

```diff
@@ -110,9 +114,54 @@ type IPCache struct {
 	deferredPrefixRelease *asyncPrefixReleaser
 }
 
+func NewIPCache(
+	lifecycle hive.Lifecycle,
+	identityAllocatorPromise promise.Promise[cache.IdentityAllocator],
+	policyRepoPromise promise.Promise[*policy.Repository],
+	datapathHandler ipcacheTypes.DatapathHandler,
+) *IPCache {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	ipcache := newIPCache(&Configuration{
+		Context:         ctx,
+		DatapathHandler: datapathHandler,
+	})
+
+	go func() {
+		policyRepo, err := policyRepoPromise.Await(ctx)
```

Comment:
Do the awaits from Start() and use HookContext so Await() cancels on start timeout.